apps/worker/src/app/workflow/usecases/run-job/run-job.usecase.ts

```diff
@@ -77,6 +77,8 @@ export class RunJob {
           job,
         })
       );
+
+      await this.jobRepository.updateStatus(job._environmentId, job._id, JobStatusEnum.COMPLETED);
```

Comment:
Why not using the use case `SetJobAsCompleted`?
```
await this.setJobAsCompleted.execute(
        SetJobAsCommand.create({
          environmentId,
          jobId,
          userId,
        })
      );
```
I get the reason to move to the the use case the completion flow instead of in the worker but if we are afraid of potential race conditions and want to directly hit the DAL we might just be masking a bit still the concurrency problem.
Unless you have discovered also that there is some flow in the `on('completed', ...)` path of BullMQ that is delaying executions.

Reply:
I think we just missed that usecase, @scopsy do you remember if there was another reason for it?